batavia/types/Str.js

```diff
@@ -233,7 +233,7 @@ String.prototype.__iadd__ = function(other) {
 };
 
 String.prototype.__isub__ = function(other) {
-    throw new batavia.builtins.NotImplementedError("String.__isub__ has not been implemented");
+    throw new batavia.builtins.TypeError("unsupported operand type(s) for -=: 'str' and '" + batavia.type_name(other) +"'")
```

Comment:
Um, I'm not sure I understand, what's the "good" behavior? Doesn't batavia.type_name() account for when the argument is null or NoneType? And why would the NotImplementedError still be left in?